project/BuildSettings.scala

```diff
@@ -0,0 +1,87 @@
+// SBT
+import sbt._
+import Keys._
+
+// sbt-assembly
+import sbtassembly._
+import sbtassembly.AssemblyKeys._
+
+/**
+  * Common settings-patterns for Snowplow apps and libraries.
+  * To enable any of these you need to explicitly add Settings value to build.sbt
+  */
+object BuildSettings {
+
+  lazy val buildSettings = Seq[Setting[_]](
+    scalacOptions := Seq(
+      "-deprecation",
+      "-encoding", "UTF-8",
+      "-feature",
+      "-unchecked",
+      "-Ywarn-dead-code",
+      "-Ywarn-inaccessible",
+      "-Ywarn-infer-any",
+      "-Ywarn-nullary-override",
+      "-Ywarn-nullary-unit",
+      "-Ywarn-numeric-widen",
+      "-language:higherKinds",
+      "-Ywarn-unused",
+      "-Ypartial-unification",
+      "-Ywarn-value-discard"
+    ),
+    javacOptions := Seq(
+      "-source", "1.8",
+      "-target", "1.8",
+      "-Xlint"
+    )
+  )
+
+  // Makes package (build) metadata available withing source code
+  lazy val scalifySettings = Seq(
+    sourceGenerators in Compile += Def.task {
+      val file = (sourceManaged in Compile).value / "settings.scala"
+      IO.write(file, """package com.snowplowanalytics.snowplow.webmodel.generated
+                       |object ProjectMetadata {
+                       |  val version = "%s"
+                       |  val name = "%s"
+                       |  val organization = "%s"
+                       |  val scalaVersion = "%s"
+                       |}
+                       |""".stripMargin.format(version.value, name.value, organization.value, scalaVersion.value))
+      Seq(file)
+    }.taskValue
+  )
+
+  // sbt-assembly settings
+  lazy val assemblySettings = Seq(
```

Comment:
Everything mention `assembly` is related to feature of "building a single file with all dependencies", "fatjar" in other words. Concept of "fatjar" is very common, but details are not relevant for now.